test/ctia/task/migration/migrate_es_stores_test.clj

```diff
@@ -74,10 +71,25 @@
   (es-index/delete! @es-conn "v0.0.0*")
   (es-index/delete! @es-conn (str (migration-index) "*")))
 
-(use-fixtures :each
-  (join-fixtures [helpers/fixture-ctia
-                  es-helpers/fixture-delete-store-indexes
-                  fixture-clean-migration]))
+(defn with-each-fixtures*
+  "Wrap this function around each deftest instead of use-fixtures
+  so the TK config can be transformed before helpers/fixture-ctia
+  starts the app."
+  [config-transformer body-fn]
+  (let [fixtures (join-fixtures [;; TODO move to after fixture-ctia. currently not
+                                 ;; possible because fixture-setup! calls `p/init!`
+                                 fixture-setup!
+                                 helpers/fixture-ctia
+                                 es-helpers/fixture-delete-store-indexes
+                                 fixture-clean-migration])]
+    (helpers/with-config-transformer*
+      config-transformer
+      #(fixtures body-fn))))
+
+(defmacro with-each-fixtures
+  "Primarily for check-migration-params-test."
+  [config-transformer & body]
+  `(with-each-fixtures* ~config-transformer #(do ~@body)))
```

Comment:
It's a nice clarification of the context of execution of each test. I am wondering if we should not go further and explicitly pass the configuration instead of a transformer in order to clarify the context of execution of a migration test (and the lack of clarity of this context is totally my bad here ;-) )

Reply:
It's kind of difficult because CTIA is usually started by a test fixture. That's what motivated this strange "transformer" helper, as well as not having to rewrite existing config "transformers" that were already present from `with-redefs`.

I think once we move to `with-app-with-config` (if ever) then we should totally pass the entire map. Otherwise, I think this approach is more composable (eg., who wins if we bind the entire config twice?).